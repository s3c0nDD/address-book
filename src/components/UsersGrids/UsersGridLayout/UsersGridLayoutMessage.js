import React  from 'react';
import PropTypes from 'prop-types';

import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';

/**
 * User grid layout message component
 * @param {object} props component's props
 * @param {object} props.className classnames object
 * @param {function} props.message text to present
 */
const UsersGridLayoutMessage = ({ className, message }) => {
  return (
    <Grid
      className={className}
      container
      justify="center"
      alignItems="center"
      spacing={4}
    >
      <Grid item>
        <Typography
          variant="h4"
          color="textSecondary"
        >
          {message}
        </Typography>
      </Grid>
    </Grid>
  );
};

export default UsersGridLayoutMessage;

UsersGridLayoutMessage.propTypes = {
  className: PropTypes.any,
  message: PropTypes.string.isRequired
};