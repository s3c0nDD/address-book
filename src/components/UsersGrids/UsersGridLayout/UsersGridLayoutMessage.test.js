import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import UsersGridLayoutMessage from './UsersGridLayoutMessage'

describe('UsersGridLayoutMessage', () => {
  let shallow;
  let wrapper;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    wrapper = shallow(
      <UsersGridLayoutMessage
        message="Hello world"
      />
    );
  });

  it('shallowly renders and matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});