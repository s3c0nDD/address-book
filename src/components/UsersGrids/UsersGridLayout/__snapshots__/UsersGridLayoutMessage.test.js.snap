// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersGridLayoutMessage shallowly renders and matches snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  justify="center"
  spacing={4}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <WithStyles(ForwardRef(Typography))
      color="textSecondary"
      variant="h4"
    >
      Hello world
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
